### DISCUSSION TITLE
能否在编译时将xray作为一个可选项而非必选项呢？

### DISCUSSION THREAD

**Discussion by zhoufla** [Openwrt-Passwall/openwrt-passwall2#615] [Q&A]
我现在只使用sing-box核心和haproxy负载均衡搭配，体验良好，xray核心基本用不到，白白占用存储空间。

**Comment by nczyw** [comment]
为啥我的passwall2，type选择sing-box，然后就完全连不上了呢.一直没搞懂，passwall2，如何使用sing-box的，只能选xray.

**Reply by zhoufla** [reply]
sb核心分流需要的资源文件跟xray不是共用的，第一次需要单独下载。

**Reply by nczyw** [reply]
啊，就是要去高级里面的地址，手动下载geoip.db和geosite.db吗

**Reply by fyngd** [reply]
手动下载 geoip.db 和 geosite.db 放到 /usr/share/singbox 目录，否则跑不起来。

**Reply by nczyw** [reply]
> 手动下载 geoip.db 和 geosite.db 放到 /usr/share/singbox 目录，否则跑不起来。

感谢，我把默认地址换了一下，以前版本的默认地址，没有这两个文件，新版本地址默认有了，现在可以正常使用了

**Reply by dtknight** [reply]
难怪我是说singbox怎么老有问题
